lib/src/ssh_signing.rs

```diff
@@ -115,7 +115,8 @@ impl SshBackend {
     pub fn new(program: OsString, allowed_signers: Option<OsString>) -> Self {
         Self {
             program,
-            allowed_signers,
+            allowed_signers: allowed_signers
+                .map(|s| crate::file_util::expand_home_path(s.to_str().unwrap()).into_os_string()),
```

Comment:
It's probably better to do path resolution in `from_settings()` because `"~"` may be a valid path in other context. It will also help remove `.unwrap()`.